src/libraries/System.Console/src/System/Console.cs

```diff
@@ -119,9 +163,54 @@ public static ConsoleKeyInfo ReadKey(bool intercept)
             return ConsolePal.ReadKey(intercept);
         }
 
-        public static TextWriter Out => EnsureInitialized(ref s_out, () => CreateOutputWriter(OpenStandardOutput()));
+        public static TextWriter Out
+        {
+            get
+            {
+                // Console.Out shouldn't be locked while holding a lock on s_syncObject.
+                // Otherwise there can be a deadlock when another thread locks these
```

Comment:
I am wondering about locking `Console.Out` in pal. It looks like locking a public object. Is that safe?

Reply:
We undid it at one point but had to put it back.  Developers depend on being able to lock on Console.Out to synchronize things like changing console color, writing, and resetting it, and not having other threads' output interleave with that.

Reply:
I see. It is an obvious source of fragility, so I assumed it is something like that. It is basically a feature :-)